### PR TITLE
Implement logistic regression

### PR DESCRIPTION
**Pull request by JJModern**
## Describe your changes
CountVectorizer, bag_of_words, train_test_split, LogisticRegression implemented. Separate Makefile command of "make logistic-regress" added.

Only needs y-value labels that will be fed from future dataframe processing by other team members.

## Checklist before requesting a review
- [x] The code runs successfully.
- [x] I ran `make lint` and have made the bulk of changes that it has requested.

```
python -m "supreme_court_predictions" --logistic-regress
oh, what up?
Creating bag of words
Logistic Regression underway 

Prediction done:  [1 1]
```

### CODE REVIEW

**Review by FedericoDM — APPROVED** [review]
😃

File: supreme_court_predictions/__main__.py [hunk 1]
```diff
@@ -46,8 +54,11 @@
         get_data()
 
     if args.clean_data:
-        clean_data()
+        # clean_data()
```
Comment (michplunkett) [inline supreme_court_predictions/__main__.py:57]: Can you uncomment this?

File: supreme_court_predictions/models/logistic.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import numpy as np
+import pandas as pd
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+
+from supreme_court_predictions.util.files import get_full_pathway
+
+
+class Logistic:
```
Comment (michplunkett) [inline supreme_court_predictions/models/logistic.py:10]: `LogisiticRegression` would make this more clear for me.

File: Makefile [hunk 3]
```diff
@@ -35,3 +35,7 @@ clean-data:
 .PHONY: describe-data
 describe-data:
 	python -m ${BASEDIR} --describe-data
+
+.PHONY: logistic-regress
+logistic-regress:
+	python -m ${BASEDIR} --logistic-regress
```
Comment (michplunkett) [inline Makefile:41]: Add a new line at the end of the file and add the `--logistic-regress` to the end of the `run` command.

File: Makefile [hunk 4]
```diff
@@ -35,3 +35,7 @@ clean-data:
 .PHONY: describe-data
 describe-data:
 	python -m ${BASEDIR} --describe-data
+
+.PHONY: logistic-regress
+logistic-regress:
```
Comment (michplunkett) [inline Makefile:40]: I think it'd be better to have it just be `logistic-regression` as the other grammatically correct way would be to put it as `regress-logistically`, I believe.

**Review by michplunkett — CHANGES_REQUESTED** [review]
Just some smöl things. Great work overall!

File: Makefile (hunk 4 above)
Comment (JJModern) [reply Makefile:40]: sounds good. can change names!

File: supreme_court_predictions/models/logistic.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+from supreme_court_predictions.util.files import get_full_pathway
+
+
+class LogisiticRegression:
+    def __init__(self):
+        self.local_path = get_full_pathway(
+            "/supreme_court_predictions/data/clean_convokit/"
+        )
+
+        self.utterances_df = pd.read_pickle(self.local_path + "utterances_df.p")
```
Comment (chanteriam) [inline supreme_court_predictions/models/logistic.py:16]: This part may not work anymore since the file names are different depending on the token aggregation:
"case_aggregations.p" for all the cases
"advocate_aggregations.p": only for the utterances of speakers who are advocates
"adversary_aggregations.p": only for the utterances of speakers who are adversaries
"judge_aggregations.p": only for the utterances of speakers who are judges

This will be updated with my PR!

**Review by chanteriam — CHANGES_REQUESTED** [review]

File: supreme_court_predictions/models/logistic.py (hunk 5 above)
Comment (chanteriam) [reply supreme_court_predictions/models/logistic.py:16]: also the local_path will need to be "/supreme_court_predictions/data/processed/"

File: Makefile [hunk 6]
```diff
@@ -35,3 +35,7 @@ clean-data:
 .PHONY: describe-data
 describe-data:
 	python -m ${BASEDIR} --describe-data
+
+.PHONY: logistic-regression
```
Comment (michplunkett) [inline Makefile:39]: Add this command to the `run` one above it, por favor.

Comment (michplunkett) [reply Makefile:39]: Could you also add it to the list of commands in the `README.md` file?

File: supreme_court_predictions/models/logistic.py (hunk 5 above)
Comment (JJModern) [reply supreme_court_predictions/models/logistic.py:16]: you're right. Right now, it just provides sample logistic regression. Can definitely improve so that it works with aggregations.p

File: Makefile (hunk 6 above)
Comment (JJModern) [reply Makefile:39]: sure
